sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -158,8 +158,9 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
       ConvertToLocalRelation,
       PropagateEmptyRelation) :+
     // The following batch should be executed after batch "Join Reorder" and "LocalRelation".
-    Batch("Check Cartesian Products", Once,
-      CheckCartesianProducts) :+
+    Batch("Check and Optimize Cartesian Products", Once,
+      CheckCartesianProducts,
+      ReorderCrossJoinOperands) :+
```

Comment:
Will reordering here break join order optimized by `CostBasedJoinReorder`?

Reply:
Because this doesn't reorder the Join order, but it just swaps the two sides of a cross join.